src/ui/components/Video/useSmartTimeAndExecutionPoint.ts

```diff
@@ -16,22 +18,34 @@ export function useSmartTimeAndExecutionPoint() {
   const pauseExecutionPoint = useAppSelector(getExecutionPoint);
   const pauseTime = useAppSelector(getTime);
   const currentTime = useAppSelector(getCurrentTime);
+  const focusWindow = useCurrentFocusWindow();
 
-  // The "current time" represents the time shown on the Timeline
-  // This is the time the user sets when scrubbing the timeline as well as when playback is active
-  //
-  // The "pause time" is a higher-fidelity version of the current time
-  // It is often accompanied by an execution point and a pause id (which can be used to inspect values in the debugger)
-  //
-  // When possible, it is better to use the "pause time" for displaying graphics
-  // because it enables us to display a more up-to-date screenshot from DOM.repaintGraphics
-  //
-  // There are situations where we should not use this value though
-  // For instance, if the user scrubs the timeline outside of the focus window, a Pause can't be created
-  // In that case, we should show the lower-fidelity playback time so the screenshot and timeline line up
-  //
-  // A heuristic for detecting this case is comparing how close the two values are
-  const preferCurrentTime = playbackState != null || Math.abs(pauseTime - currentTime) > 0.05;
+  let preferCurrentTime = false;
+  if (playbackState != null) {
+    preferCurrentTime = true;
+  } else if (
+    focusWindow &&
+    !isTimeInRegion(currentTime, focusWindow) &&
+    Math.abs(pauseTime - currentTime) > 0.05
```

Comment:
I think this change is sufficient to fix what @hbenl originally reported. (And I believe @hbenl verified that fix on this preview branch.)

I'm not _yet_ sure how to recreate the exact failure that was reported in an e2e test, but I'm going to poke at it a bit more.